python/ml4ir/base/model/calibrations/temperature_scaling.py

```diff
@@ -0,0 +1,247 @@
+"""
+Implementing temperature scaling proposed in https://github.com/gpleiss/temperature_scaling
+The tensorflow implementation and functions are inspired by
+https://www.tensorflow.org/probability/examples/Optimizers_in_TensorFlow_Probability
+"""
+import sys
+import os
+import contextlib
+import functools
+import time
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+import tensorflow_probability as tfp
+from logging import Logger
+from ml4ir.base.data.relevance_dataset import RelevanceDataset
+from ml4ir.base.model.scoring.scoring_model import ScorerBase
+
+import shutil
+
+TEMPERATURE_SCALE = 'temp_scaling_scores'
+
+
+def make_val_and_grad_fn(fn):
+    """ Returns a function that computes y = fn(*args, **kwargs) and gradients of y wrt
```

Comment:
thank you for adding docstrings @marjanhs ! It helps! 
Please ensure consistency ie., start with capital, no space between quotes and first letter.